shortlister/tournament.py

```diff
@@ -0,0 +1,96 @@
+from itertools import combinations
+from pathlib import Path
+import pickle
+from typing import Dict, List
+from readchar import readkey
+
+
+def comparison(list_to_rank, result: Dict):
+    """Starts pair comparison for a list of items.
+    The result parameter could be dictionary with existing results or empty dictionary"""
+    # Queue is a list of pair that has not been compared yet
+    queue = get_pair(list_to_rank, result)
+
+    # Loop when there are still pairs to be compared in the queue
+    while len(queue) > 0:
+        # Remove the first pair in the queue and start the comparison
+        pair = tuple(queue.pop(0))
+        print(f"{pair[0]}[1] : {pair[1]}[2]")
+
+        # Limits key choice to "u","1","2" so there is no need for exception checking
+        while True:
+            choice = readkey()
+            if choice in ["u", "1", "2"]:
```

Comment:
We need to refine the condition here to only exit the key listening loop if user presses 1 or 2, or u _and_ results is not empty. Otherwise user can keep undoing.